### PR TITLE
Fix: httpc set_options override

### LINKED ISSUE #8829: httpc is getting the wrong ipfamily by setting some options with httpc:set_option
**Describe the bug**
httpc profile with set_option to set some options would change the ipfamily from inet6 back to inet. The order of setting some options would change the ipfamily in the state of the httpc profile.

**To Reproduce**

When I am setting at first the ipfamily inet6 and then the max_session the ipfamily got changed by 5> httpc:set_option(max_sessions,100,inet6). back to inet - this is wrong. If a change the order of set_option then it would work, why?

1> observer:start().
ok
2> inets:start().
ok
3> {ok, _} = inets:start(httpc, [{profile, inet6}]).
{ok,<0.119.0>}
4> httpc:set_option(ipfamily,inet6,inet6).
ok
5> httpc:set_option(max_sessions,100,inet6).
ok
6> 

Please look into the observer:
observer -> httpc_inet6 process -> option ipfamily = inet
it's inet, not inet6 anymore!

------
Step2):

1> observer:start().
ok
2> inets:start().
ok
3> {ok, _} = inets:start(httpc, [{profile, inet6}]).
{ok,<0.119.0>}
4> httpc:set_option(max_sessions,100,inet6).
ok
5> httpc:set_option(ipfamily,inet6,inet6).
ok
6> 
Please look into the observer:
observer -> httpc_inet6 process -> option ipfamily = inet6
it's inet6

Could you also reproduce this?

**Expected behavior**
The option setting should not change the inet6 ipfamily

**Affected versions**
OTP 26.2.5.3

### PR DESCRIPTION
**Pull request by MarkoMin**
The bug was due to invalid validation of options that was acting like the user always passed `ipfamily` and `unix_socket` options, so each time when calling `set_option(s)` those options would get overridden by default values (if option values weren't present in the new options list).

Validation of `ipfamily` and `unix_socket` combination was not just buggy, but also insufficient because the httpc state could end up with invalid combination of those options which I proved in the `httpc_SUITE:invalid_ipfamily_unix_socket/1` testcase (which fails on the current code). I also believe that there was a typo in `httpc:validate_ipfamily_unix_socket/1` where `ipfamily` option got deleted twice, but `unix_socket` was not, so we could end up having two `unix_socket` entries after validation....

With the new behavior that checks current and new options `httpc` can't end up having invalid ipfamily-unix_socket combo. I also believe that new error is more informative to the user on what's going on and why validation failed.

I also removed deprecated `ipv6` handling in `httpc_manager` because `ipv6` gets translated to `ipfamily` option in `httpc:validate_options/3` so there is no way that `ipv6` entry can ever be passed to `httpc_manager`. `httpc_manager:set_options` is called only from `httpc:set_options` which validates (and translates) `ipv6` option so this should be safe.

Closes #8829

### CODE REVIEW

File: lib/inets/src/http_client/httpc.erl [hunk 1]
```diff
@@ -700,7 +700,9 @@ Sets options to be used for subsequent requests.
       DomainDesc :: string(),
       HostName :: uri_string:uri_string().
 set_options(Options, Profile) when is_atom(Profile) orelse is_pid(Profile) ->
-    case validate_options(Options) of
+    {ok, IpFamily} = get_option(ipfamily, Profile),
```
Comment (Whaileee) [inline lib/inets/src/http_client/httpc.erl:703]: When httpc:set_options(...) is called without starting httpc first we get no match `{error,inets_not_started}` on these two lines. Although calling set_options without running client is unreasonable, this case should be handled.

**Comment by MarkoMin** [comment]
`maybe` seems like a perfect fit here

File: lib/inets/src/http_client/httpc.erl (hunk 1 above)
Comment (IngelaAndin) [reply lib/inets/src/http_client/httpc.erl]: It is the application intets that is not started. If the inets application is started  you can set the options for the default profile (default client instance) should work fine.

```
1> inets:start().
ok
2> httpc:set_options([{ipfamily, inet6}]).
ok
3>

```

Comment (Whaileee) [reply lib/inets/src/http_client/httpc.erl]: I have misphrased my comment, sorry. The issue is now resolved, so I'll close this thread

**Comment by Whaileee** [comment]
Sorry for late feedback, I have backported this change and now it's a part of #9133.
Thanks for contribution.
Best Regards,
Konrad

**Pull request closed without merging**